dlib/cuda/cudnn_dlibapi.cpp

```diff
@@ -170,6 +170,18 @@ namespace dlib
             return des.get_handle();
         }
 
+        static cudnnActivationDescriptor_t clipped_relu_activation_descriptor(float coef)
+        {
+            thread_local cudnn_activation_descriptor des(CUDNN_ACTIVATION_CLIPPED_RELU, CUDNN_PROPAGATE_NAN, coef);
+            return des.get_handle();
+        }
+
+        static cudnnActivationDescriptor_t elu_activation_descriptor(float coef)
+        {
+            thread_local cudnn_activation_descriptor des(CUDNN_ACTIVATION_ELU, CUDNN_PROPAGATE_NAN, coef);
```

Comment:
Same problem here.  It just can't be `thread_local` since `coef` is ignored on subsequent calls when it's `thread_local`.  

Either the descriptor needs to be created on every call or held as a class member if descriptor creation is too expensive.

Reply:
It's really wacky to me that cuDNN works like this too.  Like WTF, it's just a single scalar.  I don't know why there is this heavy handed descriptor.  So like, unless cuDNN's version of these is somehow faster than the cuda code you wrote I would just not bother with cuDNN for these activations.

Reply:
In fact, it wouldn't surprise me if the cuDNN implementation is slower.

Reply:
Ok, I will try again by putting the scalar as a class member and benchmark them.
I also thought about writing the CUDA activations myself, but I thought that the cuDNN versions would be faster.

Also, I am not sure these activations _actually_ useful, so if you think it's not worth it, we can close this PR.
It helped me better understand how the dlib cuDNN API works, so it's already a win for me.

Reply:
¯\_(ツ)_/¯   Nothing wrong with more activations.  I would honestly just write the cuda version yourself rather than deal with cuDNN for these.

Reply:
Yes, I get your point, I guess the only advantage of having the cuDNN version is to be able to fuse them with the convolution and the bias in a single cuDNN call.
I will try to make it work with cuDNN (just because I'm curious about the performance)

Also, off-topic, there's one thing that bothers me, your shruggie is missing one forearm... ¯\_(ツ)_/¯

Reply:
Ah, it's not missing! GitHub removes it :O

Reply:
You were right, after comparing both cuDNN and my CUDA implementation, there are no noticeable differences in speed.